### PR TITLE
qt515.qt3d: fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Tested with sioyek.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/qt-5/modules/qt3d.nix [hunk 1]
```diff
@@ -1,7 +1,10 @@
-{ qtModule, qtbase, qtdeclarative }:
+{ lib, stdenv, qtModule, qtbase, qtdeclarative }:
 
-qtModule {
+qtModule ({
   pname = "qt3d";
   qtInputs = [ qtbase qtdeclarative ];
   outputs = [ "out" "dev" "bin" ];
-}
+} // lib.optionalAttrs (stdenv.isDarwin && stdenv.isAarch64) {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/modules/qt3d.nix:7]: Please just inline this. Merging the attr is not that nice and for bigger files reduces readability quite a lot.

Comment (wegank) [reply pkgs/development/libraries/qt-5/modules/qt3d.nix]: Done.

File: pkgs/development/libraries/qt-5/modules/qt3d.nix [hunk 2]
```diff
@@ -1,7 +1,9 @@
-{ qtModule, qtbase, qtdeclarative }:
+{ lib, stdenv, qtModule, qtbase, qtdeclarative }:
 
-qtModule {
+qtModule ({
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/modules/qt3d.nix:3]:
```suggestion
qtModule {
```

File: pkgs/development/libraries/qt-5/modules/qt3d.nix [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
   pname = "qt3d";
   qtInputs = [ qtbase qtdeclarative ];
   outputs = [ "out" "dev" "bin" ];
-}
+  # error: use of undeclared identifier 'stat64'
+  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) "-Dstat64=stat";
+})
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/modules/qt3d.nix:9]:
```suggestion
}
```

File: pkgs/development/libraries/qt-5/modules/qt3d.nix (hunk 2 above)
Comment (wegank) [reply pkgs/development/libraries/qt-5/modules/qt3d.nix]: Oh, my bad. Fixed.

**Pull request merged**